xarray_selafin_backend/xarray_backend.py

```diff
@@ -32,23 +42,27 @@ def _raw_indexing_method(self, key):
             time_key = key
             # And we want to select all nodes
             node_key = slice(None)
-        
-        logging.info(key)
-        print(key)
 
         # Convert time_key and node_key to ranges to handle steps and to list indices for SELAFIN reader
-        time_indices = range(*time_key.indices(self.shape[0])) if isinstance(time_key, slice) else [time_key]
-        node_indices = range(*node_key.indices(self.shape[1])) if isinstance(node_key, slice) else [node_key]
+        time_indices = (
+            range(*time_key.indices(self.shape[0]))
+            if isinstance(time_key, slice)
+            else [time_key]
+        )
```

Comment:
A ternary operator that actually needs one extra line compared to a normal if/else is a prime example of "when not to use ternary operators!":
``` python
if isinstance(time_key, slice):
    time_indices = range(*time_key.indices(self.shape[0]))
else:
    time_indices = [time_key]
```
Furthermore, I recommend avoiding using ternary operators in code that you plan to test. 

The issue with ternary operators is that they merge two execution paths into one line, leading to potentially deceptive code coverage. It’s possible that your tests may only examine one of the paths, leaving you unaware of the untested path unless you specifically notice it

Reply:
indeed. 
the problem I have before even treating the case of each key is to determine how to treat the keys themselves. 
```python
        if isinstance(key, tuple):
            time_key, node_key, plan_key = key
        else:
            # Assuming if only one key is provided, it's for the time dimension
            time_key = key
            # And we want to select all nodes
            node_key = slice(None)
            # and all plans 
            plan_key = slice(None)
``` 
this feels insufficient if I want to slice only with time or only with the nodes index, or finally only with the plans (in 3D).

Reply:
isn't there any way to recognize which is the key used?

Reply:
isn't there any way to recognize which is the key used?